tests/random_test.py

```diff
@@ -1056,6 +1056,40 @@ def testOrthogonal(self, n, shape, dtype):
         atol=tol, rtol=tol,
       )
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_p={}_shape={}"\
+       .format(p, jtu.format_shape_dtype_string(shape, dtype)),
+       "p": p,
+       "shape": shape,
+       "dtype": dtype}
+      for p in [.5, 1., 1.5, 2., 2.5]
+      for shape in [(), (5,), (10, 5)]
+      for dtype in jtu.dtypes.floating))
+  def testGnormal(self, p, shape, dtype):
+    key = self.seed_prng(0)
+    samples = random.gnormal(key, p, shape, dtype)
+    self.assertEqual(samples.shape, shape)
+    self.assertEqual(samples.dtype, dtype)
+    self._CheckKolmogorovSmirnovCDF(samples.ravel(), scipy.stats.gennorm(p).cdf)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_d={}_p={}_shape={}"\
+       .format(d, p, jtu.format_shape_dtype_string(shape, dtype)),
+       "d": d,
+       "p": p,
+       "shape": shape,
+       "dtype": dtype}
+      for d in range(4)
+      for p in [.5, 1., 1.5, 2., 2.5]
+      for shape in [(), (5,), (10, 5)]
+      for dtype in jtu.dtypes.floating))
+  def testBall(self, d, p, shape, dtype):
+    key = self.seed_prng(0)
+    samples = random.ball(key, d, p, shape, dtype)
+    self.assertEqual(samples.shape, (*shape, d))
+    self.assertEqual(samples.dtype, dtype)
+    self.assertTrue(((jnp.abs(samples) ** p).sum(-1) <= 1).all())
```

Comment:
Is there any statistic we can use to gauge the correctness of the samples? I've not looked closely at the algorithm, and this test doesn't provide a strong constraint. 

For example, am I right in thinking the radial CDF should equal the hyperspherical volume formula as a function of radius? We could compute that using gamma functions and compare it with a KS test.